provider/openstack/credentials.go

```diff
@@ -153,6 +190,8 @@ func (c OpenstackCredentials) detectCredential() (*cloud.Credential, string, str
 				CredAttrAccessKey:  creds.User,
 				CredAttrSecretKey:  creds.Secrets,
 				CredAttrTenantName: creds.TenantName,
+				CredAttrTenantID:   creds.TenantID,
+				CredAttrVersion:    strconv.Itoa(creds.Version),
```

Comment:
creds.Version defaults to 0 if not specified I think.
We should only set the CredAttrVersion if creds.Version is an actual user specified value.